drivers/user_permission_report/app/views/user_permission_report/warehouse_reports/reports/index.haml

```diff
@@ -0,0 +1,30 @@
+:ruby
+  title = _('User Permission Report')
+  content_for :title, title
+  def list_items(user, func)
+    user.access_groups.map{|g| g.public_send(func)}.flatten.map(&:name).join(', ')
```

Comment:
Let's join with `join('<br />')` if the lists contain more than a small handful of items it is impossible to read currently.

Reply:
One more thought on this.  Return it as an array, do the join in the view.  That way it can easily be re-used in the future.

Reply:
I think we'll need to do more than just line breaks:

![image](https://user-images.githubusercontent.com/407636/121196966-e9255e80-c83e-11eb-9020-7d95d844e6df.png)

When lines wrap, it's hard to tell where a new item begins. The screenshot is of line breaks with commas, but that's not so great, either. Do we want to use lists?

Reply:
List is fine. You could also apply a style so they don't have internal breaks.